ext/DifferentiationInterfaceEnzymeExt/reverse.jl

```diff
@@ -1,6 +1,15 @@
 const AutoReverseEnzyme = AutoEnzyme{Val{:reverse}}
 DI.autodiff_mode(::AutoReverseEnzyme) = DI.ReverseMode()
-DI.handles_output_type(::AutoReverseEnzyme, ::Type{<:AbstractArray}) = false
+
+struct Mutating{F}
```

Comment:
Maybe this could be given a more informative name in case it pops up in stack-traces.

Reply:
`MutatingWrapperForEnzymeReverseBackend`

Reply:
`MakeAllocatingFunctionMutatingForDifferentiationInterfaceEnzymeReverseBackendWrapper` 😜

Reply:
Maybe `MakeFunctionMutatingWrapper`?

Reply:
I split the difference with `MakeFunctionMutating`